lib/mlutil.js

```diff
@@ -155,6 +155,12 @@ function endpointTransform(transform) {
 function databaseParam(connectionParams, endpoint, separator) {
   var database = connectionParams.database;
   if (database == null) {
+    if(connectionParams.basePath) {
```

Comment:
This is surprising - is `databaseParam` really guaranteed to be called for every single call to the REST API?? 

I know there are multiple REST API endpoints that do not accept "database" as a parameter - for instance:

- https://docs.marklogic.com/REST/GET/v1/rest-apis/[name]
- https://docs.marklogic.com/REST/GET/v1/config/resources
- https://docs.marklogic.com/REST/GET/v1/config/indexes

If `databaseParam` isn't called for when those endpoints are invoked - and it seems surprising that it would be called, since those endpoints don't support the "database" parameter - then the basePath won't be applied, and the call will fail.